x-pack/legacy/plugins/ml/server/models/data_recognizer/data_recognizer.ts

```diff
@@ -669,34 +770,40 @@ export class DataRecognizer {
     });
 
     // update savedObjects results
-    Object.keys(results.kibana).forEach(category => {
-      results.kibana[category].forEach(item => {
-        const result = saveResults.savedObjects.find(o => o.id === item.id);
-        if (result !== undefined) {
-          item.exists = result.exists;
-
-          if (result.error === undefined) {
-            item.success = true;
-          } else {
-            item.success = false;
-            item.error = result.error;
+    (Object.keys(results.kibana) as Array<keyof DataRecognizerConfigResponse['kibana']>).forEach(
+      category => {
+        results.kibana[category].forEach(item => {
+          const result = saveResults.savedObjects.find(o => o.id === item.id);
+          if (result !== undefined) {
+            item.exists = result.exists;
+
+            if (result.error === undefined) {
+              item.success = true;
+            } else {
+              item.success = false;
+              item.error = result.error;
+            }
           }
-        }
-      });
-    });
+        });
+      }
+    );
   }
 
   // creates an empty results object,
   // listing each job/datafeed/savedObject with a save success boolean
-  createResultsTemplate(moduleConfig) {
-    const results = {};
+  createResultsTemplate(moduleConfig: Module): DataRecognizerConfigResponse {
+    const results: DataRecognizerConfigResponse = {} as DataRecognizerConfigResponse;
     const reducedConfig = {
       jobs: moduleConfig.jobs,
       datafeeds: moduleConfig.datafeeds,
       kibana: moduleConfig.kibana,
-    };
+    } as const;
```

Comment:
I'm not familiar with this - could you explain why it's needed?

Reply:
https://www.typescriptlang.org/docs/handbook/release-notes/typescript-3-4.html#const-assertions
But I added const assertion here by mistake, will remove